bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/model/MadokaMessage.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal.model;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.BRC1HCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+public class MadokaMessage {
+
+    private static final Logger logger = LoggerFactory.getLogger(MadokaMessage.class);
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Maximum number of bytes per message chunk, including headers
+     */
+    public static final int MAX_CHUNK_SIZE = 20;
+
+    private int messageId;
+    private Map<Integer, MadokaValue> values;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private MadokaMessage() {
+        values = new HashMap<>();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    public static byte[] createRequest(BRC1HCommand command, MadokaValue... parameters) {
+
+        byte[] request = ArrayUtils.EMPTY_BYTE_ARRAY;
+
+        // Message Length - Computed in the end
+        request = ArrayUtils.add(request, (byte) 0);
+        request = ArrayUtils.add(request, (byte) 0);
+
+        // Command ID, coded on 3 bytes
+        request = ArrayUtils.add(request, (byte) 0);
+        ByteBuffer bb = ByteBuffer.allocate(2);
+        bb.putShort((short) command.getCommandId());
+        request = ArrayUtils.addAll(request, bb.array());
```

Comment:
Please don't use a ByteBuffer for such a trivial operation. Just use bitwise operators instead.
Another option which would look a bit better would be to build the message bytes through a DataOutputStream.
```
ByteArrayOutputStream output = new ByteArrayOutputStream();
DataOutputStream request = new DataOutputStream(output);
request.writeByte(0);
request.writeByte(0);
request.writeByte(0);
request.writeShort(command.getCommandId());
...
byte[] ret = output.toByteArray();
ret[1] = (byte) (request.length -1);
return ret;
```